src/main/java/com/cosmian/jna/findex/Findex.java

```diff
@@ -183,20 +135,21 @@ public static List<byte[]> search(byte[] keyK, byte[] label, Word[] words, int l
         ObjectMapper mapper = new ObjectMapper();
 
         // Findex master keys
-        if (keyK == null) {
```

Comment:
You didn't check that `label` is not null? But in `upsert` you didn't `null` check anything… Maybe it's not useful?

Reply:
Yes, this is somewhat inconsistent. Adding null checks on all parameters of all functions would make the code extremely heavy. Null check are performed on the FFI through, where things are more likely to go wrong.